### DISCUSSION TITLE
Add support for Bunny.net Stream in the Video widget

### DISCUSSION THREAD

**Discussion by talopy** [elementor/elementor#22794] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Bunny.net is widely recognized as a preferred provider for VOD (Video-on-Demand) videos. They offer two options for users to share their videos: HLS playlists and embedded iframes. Unfortunately, it appears that Elementor restricts iframe support to only Vimeo, YouTube, and Dailymotion, while relying on the self-hosted option for direct play links.

### Describe the solution you'd like

It would be great to have the option of directly adding bunny.net to the list of supported platforms in Elementor or allowing the use of custom iframe URLs without the need for the custom HTML widget. This would provide more flexibility and convenience for users when embedding videos from bunny.net.

### Describe alternatives you've considered

I have the option to utilize their MP4 fallbacks or HLS playlists with Elementor Pro. I conducted tests using developer tools to verify that changing the source during preview displays the embed correctly, and it indeed does. While I could modify the source code for a demonstration, I believe this change is straightforward to implement.

### Additional context

Direct Play URL: https://iframe.mediadelivery.net/play/90813/023136c7-cd84-468c-8514-71088232fa1f
Embed Iframe URL: https://iframe.mediadelivery.net/embed/90813/023136c7-cd84-468c-8514-71088232fa1f?autoplay=true
HLS Playlist: https://vz-21bd3842-779.b-cdn.net/023136c7-cd84-468c-8514-71088232fa1f/playlist.m3u8
Thumbnail URL: https://vz-21bd3842-779.b-cdn.net/023136c7-cd84-468c-8514-71088232fa1f/thumbnail.jpg
Thumbnail Animated URL: https://vz-21bd3842-779.b-cdn.net/023136c7-cd84-468c-8514-71088232fa1f/preview.webp
MP4 fallback: https://vz-21bd3842-779.b-cdn.net/023136c7-cd84-468c-8514-71088232fa1f/play_720p.mp4

HTML Embed Code:
```
<div style="position: relative; padding-top: 56.25%;"><iframe src="https://iframe.mediadelivery.net/embed/90813/023136c7-cd84-468c-8514-71088232fa1f?autoplay=true" loading="lazy" style="border: none; position: absolute; top: 0; height: 100%; width: 100%;" allow="accelerometer; gyroscope; autoplay; encrypted-media; picture-in-picture;" allowfullscreen="true"></iframe></div>
```

**Comment by jbaldwinNteract** [comment]
HLS isn't really a viable alternative as it only offers 720p and desktop Chrome doesn't support it. That doesn't work for us, so we need direct play support.

**Comment by jomarieminney** [comment]
This is particularly important for our clients who are predominantly in the family & domestic violence sector, for whom YouTube is a security risk (because it shows videos in browsing history). I would also very much love to see Bunny added as a native option so we don't have to use hacky workarounds.

**Comment by baskettcase** [comment]
Yes please. I just ran into this today :(

**Comment by thebluelynx** [comment]
Yes. The html widget doesn't seem to support the iframe code provided by Bunny.net.   This should be a very simple fix.

**Comment by Shepdesign** [comment]
Running into this issue now, let's add direct support for Bunny.net videos please!

**Comment by mojcasun** [comment]
One more vote for direct support for Bunny.net videos!

**Comment by KHarbourB** [comment]
And another need urgently.

**Comment by lukehsuhao** [comment]
vote +1

**Comment by CaseFixedWebdesign** [comment]
Yep, need this!

**Comment by mwatkins197** [comment]
I would like to see this too.  Another Vote.

**Comment by FITZ-Flo** [comment]
+1

**Comment by kemopup** [comment]
Would absolutely love this. Would solve a lot of problems.

**Comment by secondsky** [comment]
I had to develop a own small plugin to implement token support for bunny (this should be taken care off in the same development process to cover the security side of the implementation). It uses shortcodes and ajax loading to circumvent caching issues with the tokens. 
So would be great if we see a proper implementation.

**Reply by kemopup** [reply]
> I had to develop a own small plugin to implement token support for bunny (this should be taken care off in the same development process to cover the security side of the implementation). It uses shortcodes and ajax loading to circumvent caching issues with the tokens. 
> So would be great if we see a proper implementation.

Would you share this plugin??

**Reply by liseraz** [reply]
> I had to develop a own small plugin to implement token support for bunny (this should be taken care off in the same development process to cover the security side of the implementation). It uses shortcodes and ajax loading to circumvent caching issues with the tokens. So would be great if we see a proper implementation.

Hi! I would love to get your plugin, if possible. We have all our movies on Bunny and have to use Elementor Video Playlist to create our playlists. But then the Bunny player wont start, as it needs to be played using the embed code. Have you got a solution for this? Thanks!

**Comment by paking** [comment]
+1

**Comment by drluisbarrios** [comment]
thanks, your code, it was helpful 4ME

**Comment by willyell** [comment]
I host hundreds of videos on bunny.net stream and the fact that Elementor does not support bunny streams natively is a big limitation as Im forced to use other solutions.

**Comment by Stz000** [comment]
It will be much appreciated by many users for sure! Thank you! +1

**Comment by mahmudurrahmanlabib** [comment]
+1

**Comment by logie-ai** [comment]
+1

**Comment by willyell** [comment]
Hi There



One quick trick is you can enable 'Play Direct URL' in bunny.net under library -> security

or enable your domain under allowed domains provided your server is passing headers



let me know how it goes

Will






---- On Wed, 28 May 2025 13:44:52 +0200 Abhishek Bhikule ***@***.***> wrote ---





https://github.com/talopy
 I am having trouble viewing my Thumbnail URL, it just says 403 forbidden on the screen.
 URL: https://vz-f9672ed3-d10.b-cdn.net/13530e19-ff52-4f20-a422-0075cccd73d4/thumbnail.jpg
Whereas your's Thumbnail URL works absolutely fine and showing the thumbnail.

Am I missing any settings ? please help

**Comment by aksld** [comment]
Has anyone found a solution for integrating a Bunny video?

**Reply by logie-ai** [reply]
Yes, I have. You need to upload it to your storage, not your streaming. And then use the link in Elementor, it'll show the video with a standard player so you don't enjoy the special player features bunny has but you enjoy the speed and price of their cdn. Also, I was posting vertical videos and had to overwrite the css to show it as a vertical video otherwise Elementor forces it to be horizontal. 
Last but not least, you can still use streaming videos by enabling the mp4 fallback feature in settings and using a special url you build with the video id and adding the resolution to the end of the file. Here's a step by step guide by bunny:

https://support.bunny.net/hc/en-us/articles/5154991563026-How-to-retrieve-an-MP4-URL-from-Stream

However, you might get frustrated that it doesn't work. The reason is: it'll only work on NEW files you upload to bunny AFTER you change the settings, so change your settings on bunny and then upload a test video and create the fallback mp4 url for it using the video id. 

Took us a few months to get this to work but it finally did and it's perfect because now we can fully automate our video postings on our websites 🙂

**Comment by elimae17** [comment]
+1

**Comment by tobias992** [comment]
It has been open for more than two years. This is further proof that the Elementor development team is a joke. Working against their community and customers is their main business.

**Comment by PeteMuzz** [comment]
+1 for this. Can't believe this simple request is two years old and hasn't been implemented.

**Comment by testy-cool** [comment]
I would like to request that you not implement Bunny.net support. I've had issues with this for the past two years and I am very happy with the current situation. Thank you and keep it up! 🙏

**Reply by aksld** [reply]
What problems have you encountered?
